fixtures/dom/src/components/fixtures/error-handling/index.js

```diff
@@ -42,13 +46,45 @@ class Example extends React.Component {
   render() {
     return (
       <div>
-        <ErrorBoundary error={this.props.error} key={this.state.key} />
         <button onClick={this.restart}>Reset</button>
+        <ErrorBoundary
+          buttonText={this.props.buttonText}
+          throws={this.props.throws}
+          key={this.state.key}
+        />
       </div>
     );
   }
 }
 
+class TriggerErrorAndCatch extends React.Component {
+  container = document.createElement('div');
+
+  triggerErrorAndCatch = () => {
+    // Use setImmediate so that the render is syncrhonous
```

Comment:
Also just realized this isn't an amazing explanation. Without the `setImmediate`, the update is batched because it's inside an event handler. With the `setImmediate`, the update is deferred until the next tick, but within that tick, the `setState` is synchronous.